test/evaluation/predicateRelatedFieldEvaluationHandler.spec.ts

```diff
@@ -0,0 +1,49 @@
+import fs from 'fs';
+
+import { Optionality, PresentationDefinition } from '@sphereon/pe-models';
+
+import { Checked, Status } from '../../lib';
+import { EvaluationClient } from "../../lib/evaluation/evaluationClient";
+import { HandlerCheckResult } from "../../lib/evaluation/handlerCheckResult";
+
+function getFile(path: string) {
+    return JSON.parse(fs.readFileSync(path, 'utf-8'));
+}
+
+describe('evaluate', () => {
+
+    it('should return error if value of predicate is not one of [required, preferred]', function () {
+        const pdSchema = getFile('./test/dif_pe_examples/pd/pd-simple-schema-age-predicate.json').presentation_definition;
+        pdSchema.input_descriptors[0].constraints.fields[0].predicate = "necessary";
+        const validPredicates = ['required', 'preferred'];
+        const exceptionResults = [];
+        pdSchema.input_descriptors.forEach(id => {
+            if (id.constraints) {
+                id.constraints.fields.forEach(f => {
+                    if (!validPredicates.includes(f.predicate)) {
+                        exceptionResults.push(new Checked('root.input_descriptor', Status.ERROR, 'predicate value should be one of these values: [\'required\', \'preferred\']'));
+                    }
+                })
+            }
+        })
+        expect(exceptionResults.length).toEqual(1);
+        expect(exceptionResults[0]).toEqual(new Checked('root.input_descriptor', Status.ERROR, 'predicate value should be one of these values: [\'required\', \'preferred\']'));
+    });
```

Comment:
we are not testing evaluation method in this test case. No part of the source code seem to be tested in this test case. We should remove it.